shimming-toolbox/shimmingtoolbox/cli/b0shim.py

```diff
@@ -1460,7 +1460,152 @@ def max_intensity(fname_input, fname_mask, fname_output, verbose):
+        raise ValueError("Custom-cl output format is only compatible with slicewise input format")
+    if o_format == 'slicewise' and i_format != 'slicewise':
+        raise ValueError("Slicewise output format is only compatible with slicewise input format")
+    if o_format == 'chronological' and i_format != 'chronological':
+        raise ValueError("Chronological output format is only compatible with chronological input format")
+
+    # Prepare the output
+    create_output_dir(fname_output, is_file=True)
+
+    with open(fname_input, 'r') as f:
+        coefs1 = []
+        for i_line, line in enumerate(f):
+            list_line = line.strip('\n').split(',')
+            temp = []
+            for i, value in enumerate(list_line):
+                if value.strip(' ') != '':
+                    temp.append(float(value.strip()))
+            coefs1.append(temp)
+        coefs1 = np.array(coefs1)
+    with open(fname_input2, 'r') as f:
+        coefs2 = []
+        for i_line, line in enumerate(f):
+            list_line = line.strip('\n').split(',')
+            temp = []
+            for i, value in enumerate(list_line):
+                if value.strip(' ') != '':
+                    temp.append(float(value.strip()))
+            coefs2.append(temp)
+        coefs2 = np.array(coefs2)
+
+    if o_format == 'custom_cl':
+        if coefs1.shape[1] != 9 or coefs2.shape != 9:
+            raise ValueError("The number of channels in one of the text files must be 9 for the custom-cl format")
+
+    if coefs1.shape == coefs2.shape:
+        coefs = coefs1 + coefs2
+    else:
+        # Same number of slices or change of shims but not the same number of channels
+        # We assume the first channels are the same
+        logger.debug("Not the same number of channels in both text files")
+        if coefs1.shape[0] == coefs2.shape[0]:
+            n_shims = coefs1.shape[0]
+            n_channels1 = coefs1.shape[1]
+            n_channels2 = coefs2.shape[1]
+            coefs = np.zeros((n_shims, max(n_channels1, n_channels1)))
+            for i_shim in range(n_shims):
+                if n_channels1 > n_channels2:
+                    for i_channel in range(n_channels1):
+                        if i_channel < n_channels2:
+                            coefs[i_shim][i_channel] = coefs1[i_shim][i_channel] + coefs2[i_shim][i_channel]
+                        else:
+                            coefs[i_shim][i_channel] = coefs1[i_shim][i_channel]
+                else:
+                    for i_channel in range(n_channels2):
+                        if i_channel < n_channels1:
+                            coefs[i_shim][i_channel] = coefs1[i_shim][i_channel] + coefs2[i_shim][i_channel]
+                        else:
+                            coefs[i_shim][i_channel] = coefs2[i_shim][i_channel]
+        else:
+            raise ValueError("The number of shim events is not the same in both text files")
```

Comment:
I understand it is more practical for this specific use, but I feel that in a broader view of how it will be used, it is better to make one function convert text file formats and one function combine those text files

Reply:
I made 2 CLIs, one to add coefficients and one to convert between formats ([8d65241](https://github.com/shimming-toolbox/shimming-toolbox/pull/550/commits/8d652415419d811fd9a2ed737a886453b9212875))